aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+        )
+        se3_pose = get_a_tform_b(frame_tree_snapshot, a, b)
+        pos = se3_pose.get_translation()
+        quat = se3_pose.rotation.normalize()
+        return sp.SE3(quat.to_matrix(), pos)
+
+    def _get_body_T_headcam(self, frame_tree_snapshot_head):
+        raise RuntimeError("This method is not used anymore. Please use get_spot_a_T_b")
+
+    def _get_spotWorld_T_handcam(
+        self, frame_tree_snapshot_hand, spot_frame: str = "vision"
+    ):
+        if spot_frame != "vision" and spot_frame != "odom":
+            raise ValueError("spot_frame should be either vision or odom")
+        spot_world_frame = spot_frame
+
+        hand_bd_wrist_T_handcam_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "hand_color_image_sensor"
+            ).parent_tform_child
+        )
+        hand_mn_wrist_T_handcam = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_wrist_T_handcam_dict
+        )
+
+        hand_bd_body_T_wrist_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "arm0.link_wr1"
+            ).parent_tform_child
+        )
+        hand_mn_body_T_wrist = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_wrist_dict
+        )
+
+        hand_bd_body_T_spotWorld_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                spot_world_frame
+            ).parent_tform_child
+        )
+        hand_mn_body_T_spotWorld = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_spotWorld_dict
+        )
+        hand_mn_spotWorld_T_body = hand_mn_body_T_spotWorld.inverted()
+
+        hand_mn_body_T_handcam = hand_mn_body_T_wrist @ hand_mn_wrist_T_handcam
+        hand_mn_spotWorld_T_handcam = hand_mn_spotWorld_T_body @ hand_mn_body_T_handcam
+
+        return hand_mn_spotWorld_T_handcam
+
+    def _get_spotWorld_T_headcam(
+        self, frame_tree_snapshot_head, use_vision_as_world: bool = True
+    ):
+        raise RuntimeError("This method is not used anymore. Please use get_spot_a_T_b")
+
+    def get_avg_spotWorld_T_marker_HAND(
+        self,
+        spot_frame: str = "vision",
+        data_size_for_avg: int = 10,
+        filter_dist: float = 2.2,
+    ):
+        """
+        Returns the average transformation of spot world frame to marker frame
+
+        We get a camera_T_marker for each frame in which marker is detected.
+        Depending on the frame rate of image capture, multiple frames may have captured the marker.
+        Averaging all transforms would be best way to compensate for any noise that may exist in any frame's detections
+        camera_T_marker is used to compute spot_T_marker[i] and thus spotWorld_T_marker[i].
+        Then we average all spotWorld_T-marker to find average marker pose wrt spotWorld.
+        """
+        if spot_frame != "vision" and spot_frame != "odom":
```

Comment:
Instead of "vision" or "odom" these should probably be constants being accessed as part of Spot class?

Reply:
added them as frame names inside ros_frames.yaml file

Reply:
@zephirefaith let's add a nested name structure inside ros_frame_names.yaml where we can describe all generic frames and specific frames too .. Eg:

```
rf.SPOT.WORLD_ODOM
rf.SPOT.BASE
rf.SPOT.ARM.WRIST_0
rf.ARIA.CPF
rf.ARIA.DEVICE
rf.ARIA_WORLD
rf,.MARKER
...
```